File: test/c/bufferclean.c
#include <stdio.h>

int main() {
	char tape[32000] = {0};
	char* ptr = tape;

	while (*ptr) {
		putchar(*ptr);
		}
	*ptr = getchar();
	putchar(*ptr);
	*ptr = getchar();
	*ptr = getchar();
	*ptr = getchar();
	putchar(*ptr);
	putchar(*ptr);

	return 0;
}

File: test/py/bufferclean.py
#!/usr/bin/env python3
tape = [0] * 32000
ptr = 0
while tape[ptr] != 0:
	print(chr(tape[ptr]), end='')
	pass
tape[ptr] = ord(input()[0])
print(chr(tape[ptr]), end='')
tape[ptr] = ord(input()[0])
tape[ptr] = ord(input()[0])
tape[ptr] = ord(input()[0])
print(chr(tape[ptr]), end='')
print(chr(tape[ptr]), end='')
